pkg/controller/disruption/disruption.go

```diff
@@ -336,7 +338,8 @@ func (r *ReconcileMachineDisruption) updateMachineDisruptionBudgetStatus(
 		ExpectedMachines:          expectedCount,
 		MachineDisruptionsAllowed: disruptionsAllowed,
 		DisruptedMachines:         disruptedMachines,
-		ObservedGeneration:        mdb.Generation,
+		// We want that the ObservedGeneration and Generation fields will be the same after the MDB update
+		ObservedGeneration: mdb.Generation + 1,
```

Comment:
this does not seems related to the title of this commit

Reply:
@enxebre I encounter the problem with it when I tested integration, can I just add a better description under the commit instead of creating a new one?

Reply:
Can you explain this? This doesn't make sense to me. You'd typically use the observed generation to detect changes to the spec, but in that case you'd set it to whatever generation you acted on, not just increment it.

Reply:
Sure, `Generation` field updated each time on the object update, status can be updated by MHC or by MDB controllers, but in the case when it updated by MDB, I want to have `ObservedGeneration` equals to `Generation` otherwise on each update of status by MDB `ObservedGeneration` will be different from `Generation` and in this case check https://github.com/openshift/machine-api-operator/pull/273/files#diff-614d58186947ca2e4e215d42c496d72eR383 will always fail, but we want to fail it only when MDB CR updated by MHC controller and not by MDB.

Reply:
This is definitely wrong since the MDB CR is updated 8 times a second (it can be even higher depending on your CPU). Everytime you update the CR, it triggers another reconciling loop since `ObservedGeneration` is always different.

Reply:
Even with this change the controller reconciles the CR all over again since the CR is always different. It always gets updated and reconciled again.

Reply:
It is the good point will check the code again

Reply:
@ingvagabund ah but we have this check before
```
if mdb.Status.CurrentHealthy == currentHealthy &&
		mdb.Status.DesiredHealthy == desiredHealthy &&
		mdb.Status.ExpectedMachines == expectedCount &&
		mdb.Status.MachineDisruptionsAllowed == disruptionsAllowed &&
		reflect.DeepEqual(mdb.Status.DisruptedMachines, disruptedMachines) &&
		mdb.Status.ObservedGeneration == mdb.Generation {
		return nil
	}
```
so it should be ok

Reply:
It's not ok. I can record a video that shows you mdb is reconciling like crazy. Though, no matter if it's `ObservedGeneration: mdb.Generation` or `ObservedGeneration: mdb.Generation + 1`, it's still looping all over again. So, this actually does not break/fix anything since it's already broken.

Reply:
it fixes the problem with the check under https://github.com/openshift/machine-api-operator/pull/273/files#diff-614d58186947ca2e4e215d42c496d72eR383 I wasn't aware that it reconciles like crazy, can you give a link on the video I will take a look

Reply:
This fixes the case: https://github.com/openshift/machine-api-operator/pull/350

Reply:
No need to `ObservedGeneration: mdb.Generation + 1` at all. The reason for the endless reconciliation loop is the call to `client.Update(...)`, which always bumps object's `Generation` field. The right way to update the status is to call `client.Status().Update(...)`, which will just updated the status without changing the object's `Generation` field.

Reply:
done

Reply:
Good catch and thanks for the fix.